modules/drivers/mongo/test/metabase/driver/mongo_test.clj

```diff
@@ -326,3 +330,49 @@
                                           {"$sort" {"_id" 1}}
                                           {"$project" {"_id" false, "count" true}}]
                             :collection  "venues"}})))))))
+
+(defn- create-database-from-row-maps! [database-name collection-name row-maps]
+  (or (db/select-one Database :engine "mongo", :name database-name)
+      (let [dbdef {:database-name database-name}]
+        ;; destroy Mongo database if it already exists.
+        (tx/destroy-db! :mongo dbdef)
+        (let [details (tx/dbdef->connection-details :mongo :db dbdef)]
+          ;; load rows
+          (mongo.u/with-mongo-connection [conn details]
+            (doseq [[i row] (map-indexed vector row-maps)
+                    :let    [row (assoc row :_id (inc i))]]
+              (try
+                (mc/insert conn collection-name row)
+                (catch Throwable e
+                  (throw (ex-info (format "Error inserting row: %s" (ex-message e))
+                                  {:database database-name, :collection collection-name, :details details, :row row}
+                                  e)))))
+            (println (format "Inserted %d rows into %s collection %s."
+                             (count row-maps) (pr-str database-name) (pr-str collection-name))))
+          ;; now sync the Database.
+          (let [db (db/insert! Database {:name database-name, :engine "mongo", :details details})]
+            (sync/sync-database! db)
+            db)))))
+
+(defn- json-from-file [^String filename]
+  (with-open [rdr (java.io.FileReader. (java.io.File. filename))]
+    (json/parse-stream rdr true)))
+
+(defn- array-fields-db []
+  (create-database-from-row-maps!
+   "test-16299"
+   "coll"
+   (json-from-file "modules/drivers/mongo/test/metabase/driver/array-fields.json")))
```

Comment:
Some test tooling to create one-off MongoDB databases/collections from JSON files for testing stuff like this